### DISCUSSION TITLE
What is the pickit code for items with replenish life, + defense roll, or cast skill lvl x - x roll when struck?

### DISCUSSION THREAD

**Discussion by danmm55** [blizzhackers/pickits#9] [Q&A about pickit codes]
What is the pickit code for items with replenish life, + defense roll (as in +10-20 defense), or cast skill lvl x - x roll when struck? (cast lvl 13-19 meteor when struck)

**Comment by theBGuy** [comment]
This is not an issue

**Comment by mf022** [comment]
> replenish life

https://github.com/blizzhackers/kolbot/blob/master/d2bs/kolbot/libs/NTItemAlias.dbl#L893
[hpregen]


> plus defense

https://github.com/blizzhackers/kolbot/blob/master/d2bs/kolbot/libs/NTItemAlias.dbl#L848
[plusdefense] >= 10


> cast skill lvl x - x roll when struck

check <https://github.com/blizzhackers/pickits/blob/master/NipGuide.md/#additional-skill-effect>
https://github.com/blizzhackers/kolbot/blob/master/d2bs/kolbot/libs/NTItemAlias.dbl#L1311
[itemskillongethit] == 56

**Comment by danmm55** [comment]
awesome thanks. what about regenerate mana? i forgot about that one. for caster amys and caster boots mainly?
oh yeah how can i pick items that roll lvl 13-19 meteor when struck? like rising sun?

**Comment by danmm55** [comment]
oh yeah is there any code for damage or attack rating to demons/undead?

**Comment by mf022** [comment]
> how can i pick items that roll lvl 13-19 meteor when struck? like rising sun?

idk, maybe some testing is needed, but 2% chance is too low, like in case of The Rising Sun amulet.

also you should consider:
* Rares Get 6 affix, 3 prefix and 3 suffix
* Crafts Get 4 affix + craft mods, still limited to 3 prefix or suffix max


> what about regenerate mana?

the answer always will be in .../libs/NTItemAlias.dbl file.
https://github.com/blizzhackers/kolbot/blob/master/d2bs/kolbot/libs/NTItemAlias.dbl#L841-L842
[manarecovery] or [manarecoverybonus]



> damage or attack rating to demons/undead?

https://github.com/blizzhackers/kolbot/blob/master/d2bs/kolbot/libs/NTItemAlias.dbl#L1258-L1259
[attackvsmontype]
[damagevsmontype]
you have an example on https://github.com/blizzhackers/pickits/blob/master/custom/kolbot.nip#L19
